tests/xtensor/test_shape.py

```diff
@@ -369,3 +380,195 @@ def test_squeeze_errors():
     fn2 = xr_function([x2], y2)
     with pytest.raises(Exception):
         fn2(x2_test)
+
+
+def test_expand_dims_explicit():
```

Comment:
There's a lot of redundancy in these tests, nothing is gained from testing 1D, 2D, 3D...

Reply:
Let me push back on this argument. This test will run thousands to times in a year, small costs add up. For some people a nice test suite takes milliseconds to run for other under  a few hours. Speed is subjective

I can't conceive of a scenario where expand_dims would fail for 2d but work for 3d or vice versa. If so why not test up to nd?

My comment was also a bit more broad, expand_dims is not such a complex operation that I think we need to test to exhausition. Right now I think it's the Op with most checks after indexing (which is actually really complex). 

We also know AI suggestions tend to err on the side of verbosity not conciseness and I suppose that translates to the tests